docs/zeebe/dynamic_scaling.md

```diff
@@ -0,0 +1,209 @@
+2. **Bootstrap Completion Signal**: Each new partition signals completion of bootstrap to partition 1
+3. **Distributed Deployment**: When a partition is bootstrapped, partition 1 distributes queued deployment commands to the partitions (including existing ones)
+4. **Acknowledgment**: Receiving partitions acknowledge successful processing of deployment commands
+5. **Retry Logic**: Failed distributions are automatically retried with exponential backoff
+
+### Scaling steps
+
+1. A client sends a [ClusterConfigurationManagementRequest](../../zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/api/ClusterConfigurationManagementApi.java) as a HTTP PATCH request at `/actuator/cluster/` on the management port.
+
+   If new brokers need to be added, it can be done in the same command, by specifying the ids of the brokers to add:
+   - with brokers:
+
+   ```bash
+   curl -X 'PATCH' \
+   'http://localhost:9600/actuator/cluster' \
+   -H 'accept: application/json' \
+   -H 'Content-Type: application/json' \
+   -d '{  
+   "brokers": { 
+   "add": [3,4,5]
+   },
+   "partitions": {
+   "count": 6, 
+   "replicationFactor": 3
+   }
+   }' | jq
+   ```
+
+   - without brokers
+
+   ```bash
+   curl -X 'PATCH' \
+   'http://localhost:9600/actuator/cluster' \
+   -H 'accept: application/json' \
+   -H 'Content-Type: application/json' \
+   -d '{  
+   "brokers": { 
+   "add": [3,4,5]
+   },
+   "partitions": {
+   "count": 6, 
+   "replicationFactor": 3
+   }
+   }' | 
+   ```
+2. The [ClusterChangeCoordinator](../../zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/changes/ConfigurationChangeCoordinator.java) generates the `ClusterChangeOperation` needed to reach the requested result.
+   For a pure scaling operation, the generated `ClusterChangeOperation` will be:
+   - `StartPartitionScaleUp` to Partition 1
+   - `PartitionBootstrapOperation` to all new partitions
+   - `AwaitRedistributionCompletion` to the coordinator node 1
+
+   Once generated, the coordinator node will execute one operation at a time, on the correct node.
+
+   **Partition ID Requirements**: New partitions must have contiguous IDs (e.g., if you currently have partitions 1-3, you can only scale to 4, 5, 6, etc.). Non-contiguous partition IDs are rejected to maintain compatibility with existing routing logic.
+
+3. `StartPartitionScaleUp`:
+
+   - Partition 1 marks that the scale up is started in the state.
+   - new partitions are added to the `desiredPartitionCount` in the state
+   - commands distribution will enqueue new commands for the new partitions, but it will not start distributing them yet.
+4. `PartitionBootstrapOperation` on the new partitions (one partition at a time, from the *lowest* to the *highest*):
+   - The partition is bootstrapped in the node with the flag `bootstrapFromSnapshot=true`.
+   - It will request the snapshot for bootstrap to partition 1.
```

Comment:
:wrench: Also mention that this snapshot contains all global resources and link to the following section that details the snapshot process.